### DISCUSSION TITLE
Tidal height calculation problem

### DISCUSSION THREAD

**Discussion by q1165600785** [pyTMD/pyTMD#317] [General]
Hi, researcher. I am using fvcom to build a regional tidal model and pyfvcom to make tidal forcing files. pyfvcom uses the UTIDE package to calculate tidal heights. Its core calculation code is   fit = np.dot(E, ap) + np.dot(np.conj(E), am)
**https://github.com/wesleybowman/UTide/blob/master/utide/_reconstruct.py**
It uses conjugate complex numbers to calculate tidal heights. 
I now want to use pytmd to calculate tidal heights as my input file  

I now want to use pytmd to calculate tidal heights as my input file. I read the code and thought that its core formula is ht.data[:] += pf[:,k]*hc.real[0,k]*np.cos(th) -pf[:,k]*hc.imag[0,k]*np.sin(th).
What confuses me is that there is no similarity between these two formulas.

I am a beginner in tidal research. Have I overlooked something?

**Comment by q1165600785** [comment]
I read the official code of aviso-fes , and the location of its calculation of tidal height is https://github.com/CNES/aviso-fes/blob/main/include/fes/tide.hpp.
I read its source code auto tide = wave->f() * ​​(wave->tide().real() * std::cos(phi) + wave->tide().imag() * std::sin(phi)); It is addition, while in pytmd it is subtraction. Why is this? Or did I miss something? Can the professor please answer my doubts?

**Comment by tsutterley** [comment]
Hi @q1165600785, the formalism for phase lags within `pyTMD` is consistent and matches the [matlab Tide Model Driver (TMD)](https://github.com/EarthAndSpaceResearch/TMD_Matlab_Toolbox_v2.5/blob/master/TMD/FUNCTIONS/harp1.m#L33-L34) and the [OSU Tidal Prediction Software (OTPS)](https://www.tpxo.net/otps).
